sdcm/nemesis.py

```diff
@@ -2043,6 +2045,70 @@ def modify_table_speculative_retry(self):
                    "'%sms'" % random.randint(1, 1000))
         self._modify_table_property(name="speculative_retry", val=random.choice(options))
 
+    def modify_table_twcs_window_size(self):
+        """ Change window size for tables with TWCS
+
+            After window size of TWCS changed, tables should be
+            reshaped. Process should not bring write amplification
+            if size of sstables in timewindow is differs significantly
+        """
+
+        def get_new_window_settings(current_size: int, current_unit: str) -> Tuple[int, str]:
+
+            if current_unit == "DAYS":
+                current_size = current_size + 1
+            elif current_unit == "HOURS":
+                if current_size // 24 > 2:
+                    current_unit = "DAYS"
+                    current_size = 3
+                else:
+                    current_size += 10
+            else:
+                if current_size // 60 > 10:
+                    current_unit = "HOURS"
+                    current_size = 11
+                else:
+                    current_size += 35
+            return current_size, current_unit
+
+        ks_cs_with_twcs = self.cluster.get_all_tables_with_twcs(self.target_node)
+        self.log.debug("All tables with TWCS %s", ks_cs_with_twcs)
+
+        if not ks_cs_with_twcs:
+            raise UnsupportedNemesis('No table found with TWCS')
+
+        ks_cs = random.choice(ks_cs_with_twcs)
+        new_size, new_unit = get_new_window_settings(
+            int(ks_cs["compaction"]["compaction_window_size"]),
+            ks_cs["compaction"]["compaction_window_unit"])
+
+        ks_cs["compaction"]["compaction_window_size"] = new_size
+        ks_cs["compaction"]["compaction_window_unit"] = new_unit
+
+        self.log.debug("New TWCS settings %s", str(ks_cs["compaction"]))
+        self._modify_table_property(name="compaction", val=ks_cs["compaction"], keyspace_table=ks_cs["name"])
+
+        self.cluster.wait_for_schema_agreement()
+        self.target_node.stop_scylla()
+
+        reshape_twcs_records = self.target_node.follow_system_log(
+            patterns=["need reshape. Starting reshape process",
+                      "Reshaping",
+                      f"Reshape {ks_cs['name']} .* Reshaped"])
+
+        self.target_node.start_scylla()
+        reshape_twcs_records = list(reshape_twcs_records)
+        if not reshape_twcs_records:
+            self.log.warning("SStables for reshape was not found. Autocompaction already"
+                             "segregate sstables by timewindows")
+        self.log.debug("Reshape log %s", reshape_twcs_records)
```

Comment:
So even if we don't see reshapes it's fine?
What about the write_amplification? Is there a way to check that it happened?

Reply:
> So even if we don't see reshapes it's fine? 

Yes, it means, that autocompaction is already segregate sstables by timewindows

> What about the write_amplification? Is there a way to check that it happened?

I am not sure it is possible  to do, while traffic is running continuously. if scylla have  some metric for that?

Reply:
I think the best way is to  have some performance tests for that